espnet/asr/chainer_backend/asr.py

```diff
@@ -104,34 +108,42 @@ def update_core(self):
         logging.info('grad norm={}'.format(grad_norm))
         if math.isnan(grad_norm):
             logging.warning('grad norm is nan. Do not update model.')
-        else:
+        elif self.count % self.accum_grad == 0:
             optimizer.update()
+            self.clean = True
```

Comment:
why don't you clean up gradients immediately? I mean how about `optimizer.target.cleargrads()` here.